File: LeetCode/Medium/0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        count = 1
        for j in range(1, len(nums)):
            if nums[i] == nums[j]:
                if count < 2:
                    i += 1
                    nums[i] = nums[j]
                count += 1
            else:
                i += 1
                nums[i] = nums[j]
                count = 1

        return i+1

File: LeetCode/Medium/0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.go
func removeDuplicates(nums []int) int {
    i := 0
    count := 0
    
    for j := 1; j < len(nums); j++ {
        if nums[i] == nums[j] {
            if count < 1 { 
                i++
                nums[i] = nums[j]
            }
            count++
        } else {
            count = 0
            i++
            nums[i] = nums[j]
        }
    }
    
    return i + 1
}